### ISSUE TITLE
URGENT Help Needed: Need to know how to get image url with file extension using fileinput

### ISSUE THREAD

**Issue by kulmon2013** [kartik-v/bootstrap-fileinput#111]
I have a scenario where i am using fileinput to upload an image. Here is the need:

1) I select an image clicking on browse button 

2) once i click on the upload button i need to retrieve the URL of that image i have selected so that i can display it in other container.

Can you please help me to resolve this

**Comment by zebub** [comment]
You can use the `fileuploaded` event and retrieve the `src` attribute of the preview `img` element using `previewId`.

Something like:

```js
$('#input-id').on('fileuploaded', function(event, data, previewId, index) {
    var src = $('#' + previewId + ' img').attr('src');
    alert(src);
});
```

**Comment by kulmon2013** [comment]
But it's nothing displayed on alert : below is my html code
<input id="image-upload" type="file" class="file" data-preview-file-type="text"> 

javascript code:

<script>
$("#image-upload").fileinput({
		previewFileType: "image",
		allowedFileTypes: ["image"],
		showCaption: false
		//uploadUrl: "/site/file-upload"
	});
$('#image-upload').on('fileuploaded', function(event, data, previewId, index) {
    	var src = $('#' + previewId + ' img').attr('src');
    	alert(src);
	});
</script>

**Comment by kulmon2013** [comment]
alert not invoked. Below is my code (HTML and JS)
<pre>
  <input id="image-upload" type="file" class="file" data-preview-file-type="text">
$("#image-upload").fileinput({
		previewFileType: "image",
		allowedFileTypes: ["image"],
		showCaption: false
		//uploadUrl: "/site/file-upload"
	});
	
	$('#image-upload').on('fileuploaded', function(event, data, previewId, index) {
    	var src = $('#' + previewId + ' img').attr('src');
    	alert(src);
	});
</pre>

**Comment by zebub** [comment]
This works for me but the `fileuploaded` event is not invoked until after the file has been succesfully uploaded. If you want the `src` attribute as soon as you click upload, you can try the `filepreupload` event instead.

However, as stated in my edit in my first comment, I'm not certain you can use the `src` attribute value to show the preview image anywhere besides the drag and drop zone.

**Comment by kulmon2013** [comment]
I am still not able to execute this. Can you please send me one example with markup and also which version of file-input plugin you are using.

**Comment by zebub** [comment]
Version 4.1.5.

```html
<input id="image-upload" type="file">
```
```js
$("#image-upload").fileinput({
            previewFileType: "image",
            allowedFileTypes: ["image"],
            showCaption: false,
            uploadUrl: "/site/file-upload"
});
$('#image-upload').on('filepreupload', function (event, data, previewId, index) {
            var src = $('#' + previewId + ' img').attr('src');
            alert(src);
});
```

Looking over your code again, I noticed a couple of things.
By giving the `input` element the class `file`, the `fileinput` script is automatically initialized for that element. Initiliazing it again with
```js
$("#image-upload").fileinput({
    ...
});
```
will probably not work and might cause unintended behaviour (presumably).

Additionally, by commenting out the `uploadUrl` key, the script will use form submission, causing the page to reload which means you cannot use events.

Again, I have to emphasize what I stated in earlier comments:
I do not know if it is possible to use the `src` attribute of the preview `img` element to show the image in another container. If it is, I do not know how. Simply creating an `img` element elsewhere with the same `src` does not seem to work.
